src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.cs

```diff
@@ -165,6 +160,29 @@ internal sealed class ImmutabilityContext {
 			}
 		}
 
+		private bool TryGetImmutableTypeInfo(
+			ITypeSymbol type,
+			out ImmutableTypeInfo info
+		) {
+			// Check for [Immutable] etc.
+			ImmutableTypeKind fromAttributes = GetImmutabilityFromAttributes( type );
+			if( fromAttributes != ImmutableTypeKind.None ) {
+				info = new ImmutableTypeInfo(
+					kind: fromAttributes,
+					type: type
+				);
+				return true;
+			}
+
+			// Check if we were otherwise told that this type is immutable
+			if( m_extraImmutableTypes.TryGetValue( type, out info ) ) {

```

Comment:
Should probably eventually look at `OriginalDefinition` as well, but will save making such adjustments for later